core/src/main/java/io/seata/core/util/IdWorker.java

```diff
@@ -0,0 +1,184 @@
+/*
+ *  Copyright 1999-2019 Seata.io Group.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.seata.core.util;
+
+/**
+ * @author funkye
+ */
+public class IdWorker {
+
+    private volatile static IdWorker idWorker = null;
+
+    /**
+     * Start time cut (2015-01-01)
+     */
+    private final long twepoch = 1546272000000L;
+
+    /**
+     * The number of bits occupied by the machine id
+     */
+    private final long workerIdBits = 5L;
```

Comment:
You can consider merge workderIdBits and datacenterIdBits. Actually they are same one.

Reply:
ok